source/extensions/http/cache/file_system_http_cache/cache_file_header_proto_util.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "envoy/http/header_map.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/extensions/filters/http/cache/cache_entry_utils.h"
+#include "source/extensions/http/cache/file_system_http_cache/cache_file_header.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+void updateProtoFromHeadersAndMetadata(CacheFileHeader& entry, const CacheFileHeader& response);
+
+CacheFileHeader protoFromHeadersAndMetadata(const Key& key,
+                                            const Http::ResponseHeaderMap& response_headers,
+                                            const ResponseMetadata& metadata);
+CacheFileTrailer protoFromTrailers(const Http::ResponseTrailerMap& response_trailers);
+
+size_t headerProtoSize(const CacheFileHeader& proto);
+
+Buffer::OwnedImpl bufferFromProto(const CacheFileHeader& proto);
+Buffer::OwnedImpl bufferFromProto(const CacheFileTrailer& proto);
+std::string serializedStringFromProto(const CacheFileHeader& proto);
```

Comment:
I agree about proto.SerializeAsString being preferable - it's here because Envoy CI enforces not using proto.SerializeAsString. I had to explicitly add cache_file_header_proto_util.cc to an exclusions file for this, which I think is preferable to adding several files as exclusions and juggling around if those usages get moved.

bufferFromProto was also added specifically for that reason, more than for clarity - it was added first, then I ended up still needing one case of serializing to not-a-buffer!